parsing/lexer.mll

```diff
@@ -861,6 +893,27 @@ and directive already_consumed = parse
             update_loc lexbuf (Some name) (line_num - 1) true 0;
             token lexbuf
       }
+  (* Lexer rule for the #syntax directive *)
+  | "syntax" [' ' '\t']+ (lowercase identchar* as mode) [' ' '\t']*
+    (lowercase identchar* as toggle) [^ '\010' '\013'] *
```

Comment:
The space between the `[^ ...]` and the `*` is weird and makes it less clear what the `*` applies to.